### PR TITLE
Change extractColumn result argument type to BaseVector

### PR DESCRIPTION
**Pull request by pramodsatya**
extractColumn result can be represented by a BaseVector instead of a VectorPtr.

### CODE REVIEW

File: velox/exec/ContainerRowSerde.cpp [hunk 1]
```diff
@@ -814,8 +814,8 @@ void ContainerRowSerde::serialize(
 void ContainerRowSerde::deserialize(
     ByteStream& in,
     vector_size_t index,
-    BaseVector* result) const {
```
Comment (aditi-pandit) [inline velox/exec/ContainerRowSerde.cpp:817]: Why is this change needed to the deserialize API ?

Comment (pramodsatya) [reply velox/exec/ContainerRowSerde.cpp]: Reverted, thanks for the suggestion! It was initially added because of the call to deserialize from extractComplexType.

File: velox/exec/RowContainer.h [hunk 2]
```diff
@@ -238,7 +238,7 @@ class RowContainer {
       int32_t numRows,
       RowColumn col,
       vector_size_t resultOffset,
-      const VectorPtr& result);
+      BaseVector& result);
```
[the remark this answers is not in the document]
Comment (pramodsatya) [reply velox/exec/RowContainer.h:241]: Thanks for the suggestion, making it const results in compilation errors and necessitates other changes, such as making setNull function const, changing extractValuesNoNulls, and the deserialize API (and other related changes would likely be required as well).
